### PR TITLE
fix: available mana without counting the implicit

### LINKED ISSUE #8352: [Task]: fix available mana without counting the implicit
### Task description

The available mana returned by the SDK counts all outputs (implicit and explicit counts), unlike base tokens which do filter them

### PR DESCRIPTION
**Pull request by cpl121**
Closes #8352 

## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/wallet/actions/getTotalWalletBalance.ts [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
 import { getBalance } from './getBalance'
+import { getManaBalance, getTotalAvailableMana } from '../../network'
+import { getWalletById } from '../../profile'
 
 export async function getTotalWalletBalance(walletId: string, accountOutputs: OutputData[]): Promise<IBalance> {
     const balances = await getBalance(walletId)
     const blockIssuanceCredits = await getBicBalance(walletId, accountOutputs)
     const totalWalletBic = Object.values(blockIssuanceCredits).reduce((acc, bic) => acc + Number(bic), 0)
+
+    const _selectedWallet = getWalletById(walletId)
```
Comment (marc2332) [inline packages/shared/lib/core/wallet/actions/getTotalWalletBalance.ts:12]: This is going to break onboarding I think, `getTotalWalletBalance` is called in buildWalletState.ts`

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: packages/shared/lib/core/wallet/actions/getTotalWalletBalance.ts (hunk 1 above)
Comment (cpl121) [reply packages/shared/lib/core/wallet/actions/getTotalWalletBalance.ts:12]: I haven't had any problems

File: packages/shared/lib/core/wallet/actions/getTotalWalletBalance.ts [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
+    let realAvailableMana = getManaBalance(balances?.mana?.available)
+    if (_selectedWallet) {
```
Comment (marc2332) [inline packages/shared/lib/core/wallet/actions/getTotalWalletBalance.ts:14]: This doesn't make sense, you still want to get the real available mana when the wallet is created in the onboarding. Imo, `getTotalWalletBalance` should be receiving the wallet instance, not the wallet id. Wallet ID were a mistake to be used as function param.

Comment (marc2332) [reply packages/shared/lib/core/wallet/actions/getTotalWalletBalance.ts:14]: There is absolutely no case where if you call getTotalWalletBalance, `_selectedWallet` should not exist, and if it didn't then the problem is elsewhere. Which in this case is that hasn't been loaded when you are in the onboarding, hence why we should be using wallet instances and not wallet ids.

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: packages/shared/lib/core/wallet/actions/getTotalWalletBalance.ts (hunk 1 above)
Comment (marc2332) [reply packages/shared/lib/core/wallet/actions/getTotalWalletBalance.ts]: This will be avoided in https://github.com/iotaledger/firefly/pull/8404 anyway

**Review by marc2332 — APPROVED** [review]

**Pull request merged**